module-2/backend/week_two.md

```diff
@@ -26,20 +32,65 @@ end
+A foreign key acts as a placeholder for data from a table that is related to the table that contains the foreign key (IE, the teacher ID instead of any other information about the teacher).  The primary key is the key used to identify a particular row of data (the teacher's teacher ID that exists on the teachers table).  The schema is a way of describing how data is organized in each table.  The schema contains information like what the column names are and what type of data each colum contains.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+A foreign key represents a primary key of another table and is used to form a relationship between two tables.  The foreign key will refer to the primary key of the table that is being referenced.  For example, each row in the teacher's table has an ID, but each row in the students tables has a teacher_id which is used to relate the two datasets.
+
 8. What are the parts of an HTTP response?
 
+A request and response.
+
+
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+
+.where - this allows you to return multiple records based on either an activerecord method or through a raw sql qualifier.
+.join - this joins two tables by relating their primary keys and foreign keys (IE, if you did Student.join(:teacher) it would return a new table that contains the actual names of the teachers for each students rather than the foreign keys.
+.find_by - this method allows you to retried a record based on a particular value (ie Student.find_by(:name 'Brandon') 
+.find_or_create_by - This method will find a record based on criteria in the same way as find by, and return a record if it exists or create one if it does not.  
+.fourty_two - retrieve the fourty-second row of data.
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+rake db:reset - fix all the things!  it drops, creates, remigrates, recreates the db schema, and seeds the database
+rake --tasks - tell me about all of the commands when I inevitably forget about them
+rake db:seed - seeding databases makes a whole bunch of things happen.  this seems like a pretty powerful command
+
 3. What two columns does `t.timestamps null: false` create in our database?
+Time stamps that have a created_at and updated_at time for each row of data.
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+A school has many teachers and teachers belong to a school
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+class Teacher
+  belongs_to: school
+ 
+class School
+  has_many: teachers
+
 6. Give an example of when you might want to store information besides ids on a join table.
+I feel like if you add things other than ids to a join table it is no longer a join table.
```

Comment:
I would go the other way. Sometimes a join table can hide from you because it has other information, but it's still a join table. Think about `invoice_items` from Black Thursday. It had quantity and unit_price as attributes, but it also held foreign keys for invoice and item.